webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -278,4 +303,161 @@ describe('TasksComponent', () => {
       expect(component.listTrackBy(0, false)).to.equal(undefined);
     });
   });
+
+  describe('lineage and breadcrumbs', () => {
+    const bettysContactDoc = {
+      _id: 'user',
+      parent: {
+        _id: 'parent',
+        name: 'CHW Bettys Area',
+      },
+    };
+    const taskDocs = [
+      { _id: '1', emission: { _id: 'e1', dueDate: '2020-10-20' }, forId: 'a', owner: 'a' },
+      { _id: '2', emission: { _id: 'e2', dueDate: '2020-10-20' }, forId: 'b', owner: 'b' },
+    ];
+    const taskLineages = [
+      {
+        _id: 'a', lineage: [
+          {name: 'Amy Johnsons Household'},
+          {name: 'St Elmos Concession'},
+          {name: 'Chattanooga Village'},
+          {name: 'CHW Bettys Area'},
+        ]
+      },
+      {
+        _id: 'b', lineage: [
+          {name: 'Amy Johnsons Household'},
+          {name: 'St Elmos Concession'},
+          {name: 'Chattanooga Village'},
+        ]
+      }
+    ];
+
+    it('it should not change the reports lineages if the connected user is online only', async() => {
+      const expectedTasks =
+        [
+          {
+            _id: 'e1',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+              'CHW Bettys Area'
+            ],
+            overdue: true,
+            owner: 'a',
+          },
+          {
+            _id: 'e2',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+            ],
+            overdue: true,
+            owner: 'b',
+          }
+        ];
+      userContactService.get.resolves(bettysContactDoc);
```

Comment:
Unit tests usually have 3 main sections:

1. Test settings and test data definition
2. Invocation of code unit that is being tested
3. Assertions

Some people likes to add comments to flag the sections for example: // GIVEN, // WHEN, // THEN. You will see that in some old android code that we have. 
But here we usually tend to add just simple spaces between to organise it and make it easier for the reader. See exampl:
```
it('.......', () => {
  const testData = [ ..... data ....];

  userContactService.get.resolves(bettysContactDoc);
  sessionService.isOnlineOnly.returns(true);
  ... other test settings ...

  // space to separate test settings and test data from unit code execution
  await new Promise(resolve => { 
      sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
      getComponent(); // <--- This is the unit code execution that we're testing
  });

  // space to separate the assertions
  expect(component.currentLevel).to.be.undefined;
  expect((<any>TasksActions.prototype.setTasksList).args).to.deep.equal([[expectedTasks]]);
});
```

Reply:
well noted